usefulUtilities/hiFiCalendar/entityScripts/showClickForTheCalendarAppClientScript.js

```diff
@@ -0,0 +1,95 @@
+//  showClickForTheCalendarAppClientScript.js
+//
+//  Created by Milad Nazeri on 7-1-2019
+//  Copyright 2019 High Fidelity, Inc.
+// 
+//  Distributed under the Apache License, Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+
+
+(function() {
+
+    var _this = null;
+    function ShouldShowClickForTheCalendarApp() {
+        _this = this;
+        
+        _this.canGetAndSetPrivateUserData = false;
+        _this.calendarBackgroundID = null;
+        _this.clickForTheCalendarAppEntityID = null;
+    }
+
+    
+    // get the entityID -> 
+    // check to see if you have private user permsissions -> 
+    // if so create the local entity ->
+    // connect onPrivateUserDataPermissionsChanged
+    var clickForTheCalendarAppDimensions = [0.75, 0.150, 0.1];
+    var clickForTheCalendarAppPosition = [0, -0.85, 0.04];
+    var clickForTheCalendarAppScriptURL = Script.resolvePath("./clickForTheCalendarAppClientScript.js");
+    var clickForTheCalendarApp = {
+        "name": "Click this sign for the calendar App",
+        "type": "Text",
+        "text": "Click here to setup\nGoogle Calendar integration",
+        "lineHeight": 0.065,
+        "dimensions": clickForTheCalendarAppDimensions,
+        "localPosition": clickForTheCalendarAppPosition,
+        "script": clickForTheCalendarAppScriptURL
+    };
+    function preload(calendarBackgroundID) {
+        _this.calendarBackgroundID = calendarBackgroundID;
+        clickForTheCalendarApp.parentID = _this.calendarBackgroundID;
+
+        _this.canGetAndSetPrivateUserData = Entities.canGetAndSetPrivateUserData();
+        _this.maybeCreateClickForTheCalendarAppEntity();
+
+        Entities.canGetAndSetPrivateUserDataChanged.connect(_this.onCanGetAndSetPrivateUserDataChanged);
+    }
+
+
+    // if you can get and set private user data, then create the local entity 
+    function maybeCreateClickForTheCalendarAppEntity() {
+        if (_this.canGetAndSetPrivateUserData) {

```

Comment:
for example, you could (and should) replace this with `if (Entities.canGetAndSetPrivateUserData()) {`

One reason why you "should" is because (even though it's unlikely) the value of the flag could change between the time you set the flag and got here.

Reply:
I think it just being cautious, but calling the function is definitely better and simpler.